worldedit-core/src/main/java/com/sk89q/worldedit/command/WorldEditCommands.java

```diff
@@ -138,6 +143,45 @@ public void report(Actor actor,
         }
     }
 
+    @Command(
+        name = "trace",
+        desc = "Trace edit actions to see what's blocking them",
+        descFooter = "This will run the given action at your placement position, then provide a single" +
+            " extent which was the first to return a failure for that action"
+    )
+    void trace(Actor actor, LocalSession localSession,
+               @Arg(desc = "The action to trace")
+                   TracingExtent.Action action,
+               @Switch(name = 'a', desc = "Print all active extents") boolean all) throws IncompleteRegionException {
+        EditSession session = localSession.createEditSession(actor, true);
+        try (EditSession ignored = session) {
+            if (action.test.test(session, localSession.getPlacementPosition(actor))) {
+                actor.printInfo(TranslatableComponent.builder("worldedit.trace.success")
+                    .args(TextComponent.of(action.toString()))
+                    .build());
+                return;
+            }
+        }
+        List<TracingExtent> tracingExtents = session.getTracingExtents();
+        assert tracingExtents != null;
```

Comment:
leftover debug?

Reply:
No, just convincing IntelliJ that it's not `null`. I can leave it out, I just don't like leaving the warnings in IntelliJ :P